packages/core/src/utils/tagName.ts

```diff
@@ -0,0 +1,37 @@
+// Utils to handle tag names for 'tds-' prefixed components
+
+export const getTagName = (el: HTMLElement): string => el.tagName.toLowerCase();
+
+export const getTagNameWithoutPrefix = (host: HTMLElement): string => {
+  const tagName = getTagName(host);
+  // regex allows both letters and digits
+  const [, tagNameWithoutPrefix = ''] = /^(tds-)([a-z0-9-]+)$/.exec(tagName) || [];
+  return tagNameWithoutPrefix;
+};
+
+export type PrefixedTagNames = Record<string, string>;
+
+const baseTagNames = ['icon', 'input', 'text-field', 'chip']; // Add all necessary base component names
```

Comment:
I have specified the components here just for testing purposes, we could instead later on create a script like `generateTagNames`. Which would derive the component names from existing components. This would ensure that our tag names are always up-to-date and reduce manual maintenance